### PR TITLE
[Compatibility](agg-state) add be_exec_version check to avoid aggregation data format changed le…

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
…ad core dump

## Proposed changes
add be_exec_version check to avoid aggregation data format changed lead core dump
```
mysql [test]>select percentile(k3, 0.1) from d_table group by grouping sets((k1),()) order by 1;
ERROR 1105 (HY000): errCode = 2, detailMessage = (hk2.dev.selectdb-in.cc)[INTERNAL_ERROR]agg state data with percentile is not supported,current_be_exec_version=7, data_be_exec_version=4, need to rebuild the data or set the be_exec_version=4 in fe.conf
```

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by hello-stephen — DISMISSED** [review]

**Comment by hello-stephen** [comment]
@HappenLee PTAL

File: be/src/olap/rowset/segment_v2/column_reader.h [hunk 1]
```diff
@@ -242,12 +244,19 @@ class ColumnReader {
 
     Status _calculate_row_ranges(const std::vector<uint32_t>& page_indexes, RowRanges* row_ranges);
 
+    Status check_be_exec_version(int read_be_exec_version) {
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_reader.h:247]: what the func do？

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**